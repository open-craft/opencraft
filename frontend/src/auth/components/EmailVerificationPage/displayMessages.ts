const messages = {
  failedVerification: {
    defaultMessage: "The verification code used is expired or doesn't exist.",
    description: ''
  },
  thankYouMessage: {
    defaultMessage: 'Thank you for verifying your email address.',
    description: ''
  },
  followUp: {
    defaultMessage:
      'Your instance should be ready within the next 2 hours. ' +
      'In the meantime, you can start customizing your instance.',
    description: ''
  },
  contact: {
    defaultMessage:
      "At any point, don't hesitate to write to us at " +
      '{contactEmail} if you have any questions or issues.',
    description: ''
  },
  emailVerified: {
    defaultMessage: 'Email verified',
    description: ''
  },
  goToConsole: {
    defaultMessage: 'Go to the customization console',
    description: ''
  }
};

export default messages;
