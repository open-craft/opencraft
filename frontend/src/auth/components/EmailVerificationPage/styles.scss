@import '~styles/theme';


.email-verification-container {
    justify-content: center;
    text-align: center;
}

i.email-verification-title-icon {
    padding-left: 15px;
}

.email-verification-console-button {
    margin-bottom: 20px;

    i {
        padding-left: 10px;
    }
}
