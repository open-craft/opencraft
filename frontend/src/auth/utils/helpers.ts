import { V2Api } from 'global/api';

export const checkAuthAndRefreshToken = async () => {
  const refreshToken = localStorage.getItem('token_refresh');
  let response = null;

  if (refreshToken) {
    try {
      response = await V2Api.authRefreshCreate({
        data: { refresh: refreshToken }
      });
    } catch (e) {
      return false;
    }
  }

  if (response) {
    localStorage.setItem('token_access', response.access);
    return true;
  }

  return false;
};
