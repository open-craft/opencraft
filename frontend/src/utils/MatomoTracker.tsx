import React, { ReactNode } from 'react';
import {
  createInstance,
  MatomoProvider,
  useMatomo
} from '@datapunt/matomo-tracker-react';
import {
  MATOMO_ALIAS_DOMAIN,
  MATOMO_BASE_URL,
  MATOMO_MY_DOMAIN,
  MATOMO_SITE_ID
} from '../global/constants';

interface MatomoTrackerProps {
  children: ReactNode;
}

interface MatomoAdditionalConfiguration {
  [propName: string]: any;
}

interface MatomoConfiguration {
  urlBase: string;
  siteId: number;
  configurations?: MatomoAdditionalConfiguration;
}

interface MatomoUserIdTrackerProps {
  userId: string;
}

/*
 * This wrapper component initializes the Matomo tracking by wrapping the children
 * elements, if the tracking is enabled by providing a valid value for the 'MATOMO_BASE_URL'
 * constant. It allows tracking page views, events and the user id, when available.
 *
 * The 'MATOMO_ALIAS_DOMAIN' constant allows the cross-domain linking of another site with
 * this site so that they are treated as the same site in Matomo.
 */
export const MatomoTracker: React.FC<MatomoTrackerProps> = (
  props: MatomoTrackerProps
) => {
  let element = props.children;
  if (MATOMO_BASE_URL) {
    const matomoConfiguration: MatomoConfiguration = {
      urlBase: MATOMO_BASE_URL,
      siteId: parseInt(MATOMO_SITE_ID, 10)
    };
    if (MATOMO_ALIAS_DOMAIN) {
      const myDomain = `*.${MATOMO_MY_DOMAIN}`;
      const aliasDomain = `*.${MATOMO_ALIAS_DOMAIN}`;
      matomoConfiguration.configurations = {
        enableCrossDomainLinking: true,
        setDomains: [myDomain, aliasDomain]
      };
    }
    element = (
      <MatomoProvider value={createInstance(matomoConfiguration)}>
        {props.children}
      </MatomoProvider>
    );
  }
  return <>{element}</>;
};

/**
 * This is a component passes the provided user id to Matomo if Matomo tracking is enabled.
 * The 'pushInstruction' function exposed by the 'useMatomo()' hook is used to do this.
 *
 * This component is needed as all the components in the registration process are classes and hence
 * the 'yseMatomo' hook cannot be directly used in those components.
 */
export const MatomoUserIdTracker: React.FC<MatomoUserIdTrackerProps> = (
  props: MatomoUserIdTrackerProps
) => {
  const { pushInstruction } = useMatomo();
  if (MATOMO_BASE_URL && props.userId) {
    pushInstruction('setUserId', props.userId);
  }
  return null;
};
