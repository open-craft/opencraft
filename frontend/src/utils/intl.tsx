import React from "react";
import { FormattedMessage, MessageDescriptor } from "react-intl";

type MessageMap = { [k: string]: Omit<MessageDescriptor, 'id'> };

interface Props<T, K extends keyof T> {
    id: K;
    messages: T;
    values?: any;
}


export const WrappedMessage = (props: Props<MessageMap, string>) => {
    const {messages, id, ...values} = props;
    return (
        <FormattedMessage  {...messages[id]} id={id} {...values}/>
    );
};
