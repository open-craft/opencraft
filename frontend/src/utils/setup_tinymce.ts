import 'tinymce/tinymce';
import 'tinymce/themes/silver';
import 'tinymce/skins/ui/oxide/skin.min.css';
import 'tinymce/skins/ui/oxide/content.min.css';
import 'tinymce/plugins/advlist';
import 'tinymce/plugins/autolink';
import 'tinymce/plugins/lists';
import 'tinymce/plugins/link';
import 'tinymce/plugins/image';
import 'tinymce/plugins/charmap';
import 'tinymce/plugins/print';
import 'tinymce/plugins/preview';
import 'tinymce/plugins/anchor';
import 'tinymce/plugins/searchreplace';
import 'tinymce/plugins/visualblocks';
import 'tinymce/plugins/code';
import 'tinymce/plugins/fullscreen';
import 'tinymce/plugins/insertdatetime';
import 'tinymce/plugins/media';
import 'tinymce/plugins/table';
import 'tinymce/plugins/paste';
import 'tinymce/plugins/help';
import 'tinymce/plugins/wordcount';
