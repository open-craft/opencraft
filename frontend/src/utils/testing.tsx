import { createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from '../global/reducers';

export const setupComponentForTesting = (
  reactContent: JSX.Element,
  storeContents = {}
) => {
  const middleware = applyMiddleware(thunk);

  const store = createStore(
    createRootReducer(createMemoryHistory()),
    storeContents,
    middleware
  );

  return renderer.create(
    <IntlProvider textComponent={React.Fragment} locale="en">
      <Provider store={store}>
        <MemoryRouter>{reactContent}</MemoryRouter>
      </Provider>
    </IntlProvider>
  );
};
