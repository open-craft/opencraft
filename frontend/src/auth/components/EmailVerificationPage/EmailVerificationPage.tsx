import * as React from 'react';
import { InstancesModel } from 'console/models';
import { connect } from 'react-redux';
import { RootState } from 'global/state';
import { NavLink } from 'react-router-dom';
import { CONTACT_US_EMAIL, ROUTES } from 'global/constants';
import { WrappedMessage } from 'utils/intl';
import { ContentPage } from 'ui/components';
import { Button, Spinner } from 'react-bootstrap';
import { performEmailActivation } from 'auth/actions';
import messages from './displayMessages';
import './styles.scss';

interface ActionProps {
  performEmailActivation: Function;
}
interface StateProps extends InstancesModel {}
interface Props extends StateProps, ActionProps {
  // Passing URL as parameter to custom page
  match: {
    params: {
      verificationCode: string;
    };
  };
  loading: boolean;
  error: string;
  succeeded: boolean;
}

@connect<{}, ActionProps, {}, Props, RootState>(
  (state: RootState) => state.loginState,
  {
    performEmailActivation
  }
)
export class EmailVerificationPage extends React.PureComponent<Props> {
  componentDidMount() {
    /**
     * Make a the email activation request as soon as component is
     * mounted and wait for server reply in loading page.
     */
    this.props.performEmailActivation(this.props.match.params.verificationCode);
  }

  public render() {
    if (this.props.loading) {
      return (
        <ContentPage title="Email Verification">
          <Spinner animation="border" role="status">
            <span className="sr-only">Loading...</span>
          </Spinner>
        </ContentPage>
      );
    }

    if (this.props.succeeded) {
      const title = (
        <span className="page-title">
          <WrappedMessage messages={messages} id="emailVerified" />
          <i className="email-verification-title-icon far fa-check-circle fa-xs" />
        </span>
      );

      return (
        <ContentPage
          title={title}
          subtitle={<WrappedMessage messages={messages} id="thankYouMessage" />}
        >
          <div className="email-verification-container">
            <p>
              <WrappedMessage messages={messages} id="followUp" />
            </p>

            <div className="email-verification-console-button text-center">
              <NavLink exact to={ROUTES.Console.HOME}>
                <Button size="lg">
                  <WrappedMessage messages={messages} id="goToConsole" />
                  <i className="fas fa-external-link-alt fa-lg" />
                </Button>
              </NavLink>
            </div>

            <p>
              <WrappedMessage
                messages={messages}
                id="contact"
                values={{
                  contactEmail: CONTACT_US_EMAIL
                }}
              />
            </p>
          </div>
        </ContentPage>
      );
    }

    return (
      <ContentPage title="Email Verification Failed">
        <p>
          <WrappedMessage messages={messages} id="failedVerification" />
        </p>
      </ContentPage>
    );
  }
}
