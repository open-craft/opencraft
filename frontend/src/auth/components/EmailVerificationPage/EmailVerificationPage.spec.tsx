import React from 'react';
import { setupComponentForTesting } from "utils/testing";
import { EmailVerificationPage } from './EmailVerificationPage';


describe("Email verification page", function() {
  const mockTokenProps = {
    params: {
      id: "verification_token"
    }
  }

  it('renders loading page', () => {
    const tree = setupComponentForTesting(
      <EmailVerificationPage
        match={mockTokenProps}
        loading={true}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders the confirmed email verification page', () => {
    const tree = setupComponentForTesting(
      <EmailVerificationPage
        match={mockTokenProps}
        loading={false}
        succeeded={true}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders the failed email verification page', () => {
    const tree = setupComponentForTesting(
      <EmailVerificationPage
        match={mockTokenProps}
        loading={false}
        succeeded={true}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
