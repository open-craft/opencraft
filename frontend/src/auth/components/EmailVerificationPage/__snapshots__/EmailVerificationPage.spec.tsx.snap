// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email verification page renders loading page 1`] = `
<div
  className="content-page"
>
  <div
    className="title-container"
  >
    <h1>
      Email Verification
    </h1>
  </div>
  <div
    className="content-page-container"
  >
    <div
      className="content-page-content row"
    >
      <div
        className="spinner-border"
        role="status"
      >
        <span
          className="sr-only"
        >
          Loading...
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Email verification page renders the confirmed email verification page 1`] = `
<div
  className="content-page"
>
  <div
    className="title-container"
  >
    <h1>
      Email Verification
    </h1>
  </div>
  <div
    className="content-page-container"
  >
    <div
      className="content-page-content row"
    >
      <div
        className="spinner-border"
        role="status"
      >
        <span
          className="sr-only"
        >
          Loading...
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Email verification page renders the failed email verification page 1`] = `
<div
  className="content-page"
>
  <div
    className="title-container"
  >
    <h1>
      Email Verification
    </h1>
  </div>
  <div
    className="content-page-container"
  >
    <div
      className="content-page-content row"
    >
      <div
        className="spinner-border"
        role="status"
      >
        <span
          className="sr-only"
        >
          Loading...
        </span>
      </div>
    </div>
  </div>
</div>
`;
